components/raftstore/src/store/fsm/peer.rs

```diff
@@ -3212,10 +3235,14 @@ where
                     }
                 }
             }
-        } else {
-            self.fsm
-                .peer
-                .execute_transfer_leader(self.ctx, msg.get_from(), peer_disk_usage, false);
+        } else if !self
+            .fsm
+            .peer
+            .maybe_reject_transfer_leader_msg(self.ctx, msg, peer_disk_usage)
+            && (!self.ctx.cfg.warmup_entry_cache_enabled()
+                || self.fsm.peer.pre_ack_transfer_leader_msg(self.ctx, msg))
```

Comment:
Maybe check for `warmup_entry_cache_enabled` inside `pre_ack_transfer_leader_msg`. As in the comment, `pre_ack_transfer_leader_msg` is a must-do item before transfer leader.

Reply:
fixed